### PR TITLE
feat: waiting to redirect from onboarding till feed is ready

### PR DESCRIPTION
**Pull request by TomDowling**
## Changes
Currently after onboarding the flicker happens as the homepage loads before the user my-feed is ready. after a few seconds it is created and the page re-renders taking the user from popular to my-feed. This fix leaves the flow the same if you haven't selected any topics in the onboarding but wait for your feed if you have selected some

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1546 #done

### CODE REVIEW

File: packages/webapp/pages/onboarding.tsx
Comment (sshanzel) [inline packages/webapp/pages/onboarding.tsx:115]: When does this happen? Why are we stopping the redirection from here?

Comment (TomDowling) [reply packages/webapp/pages/onboarding.tsx:115]: We are doing this to ensure if the user has set categories we wait for their feed to be created before redirecting them so the page doesn't just going to popular then my-feed

Comment (sshanzel) [reply packages/webapp/pages/onboarding.tsx:115]: Right. Good catch. Thanks for the clarification.

**Review by sshanzel — APPROVED** [review]
Looks good to me! Thanks for the fix!

Would appreciate to have another pair of eyes before merging as this is part of our critical flows.

@dailydotdev/web-team

File: packages/shared/src/components/onboarding/FilterOnboarding.tsx
Comment (capJavert) [inline packages/shared/src/components/onboarding/FilterOnboarding.tsx:46]: Since `setHasCategories` is optional we can early return before ever calculating `hasSelectedTopics`.

File: packages/shared/src/components/onboarding/FilterOnboarding.tsx
Comment (capJavert) [inline packages/shared/src/components/onboarding/FilterOnboarding.tsx:17]: I would make this something more generic.. eg. `onSelectTopic` and then parent component can calculate `hasCategories` from topics it receives. It would be more reusable down the line if we want to hook up something else related to topics selection. WDYT?

**Review by capJavert — COMMENTED** [review]
Few concerns over the flow for your evaluation 🙏

File: packages/shared/src/components/onboarding/FilterOnboarding.tsx
Comment (sshanzel) [reply packages/shared/src/components/onboarding/FilterOnboarding.tsx:46]: Also, after looking at the code of `onChangeSelectedTopic` I think we don't have to add another callback and just base it on the `onSelectedTopics` callback. This would also reduce the usage of `useEffect`.

File: packages/webapp/pages/onboarding.tsx
Comment (sshanzel) [inline packages/webapp/pages/onboarding.tsx:103]: Let's reserve the `on` prefix on names for functions/handlers. I think we can swap the `hasSelectedTopics` to this one.

File: packages/webapp/pages/onboarding.tsx
Comment (capJavert) [inline packages/webapp/pages/onboarding.tsx:127]: Could we not even move this to `hasSelectedTopics` callback and just redirect from there? Sorry for multiple changes, but just realised it now when you adjusted.

Comment (TomDowling) [reply packages/webapp/pages/onboarding.tsx:127]: The issue i found it wasn't ready at that point the alerts took a while to update to created / till it was ready. Doing it in hasSelectedTopics would be too early

Comment (capJavert) [reply packages/webapp/pages/onboarding.tsx:127]: Ah, of course because of update query.

**Review by capJavert — APPROVED** [review]

**Pull request merged**
